include/zephyr/net/phy.h

```diff
@@ -118,6 +119,14 @@ __subsystem struct ethphy_driver_api {
 	/** Write PHY register */
 	int (*write)(const struct device *dev, uint16_t reg_addr,
 		     uint32_t data);
+
+	/** Read PHY device register */
+	int (*read_device)(const struct device *dev, uint8_t dev_addr, uint16_t reg_addr,
+			   uint16_t *data);
```

Comment:
```suggestion
			   uint32_t *data);
```
and same for the write function, to be consistent with existing `phy_read|write` functions